lib/sqlite-adapter.js

```diff
@@ -0,0 +1,96 @@
+const sqlite3 = require('sqlite3').verbose();
+const path = require('path');
+
+module.exports = class SqliteAdapter {
+
+    static create(pluginConfig, dbName) {
+        return new this(pluginConfig, dbName);
+    }
+
+    constructor(pluginConfig, dbName) {
+        this._dbConn = null;
+        this._pluginConfig = pluginConfig;
+        this._dbName = dbName;
+    }
+
+    init() { // TODO: add exceptions handlers
+        this._dbConn = new sqlite3.Database(path.join(this._pluginConfig.path, this._dbName));
+        console.log('db connection opened');
+        this._createTable();
+        return this;
+    }
+
+    close() {
+        console.log('closing db connection');
+        return this._dbConn.close();
+    }
+
+    write(testResult) {
+        const testResultObj = this._parseTestResult(testResult);
+        const values = this._createValuesArray(testResultObj);
+        const error = this._insert(values);
```

Comment:
It's a good question - it depends on what exactly are we going to do if there's an error during the insert. I agree that error handling should go inside _insert(), but do we want to interrupt the program?

Right now, since we're writing to both data.js and sqlite.db I think it's better to throw an error\warning and continue program execution - this way we will not lose the test results and will still know that something went wrong.

But when we will get rid of writing test results to data.js, it will be better to interrupt program execution.